metaflow/datatools/s3op.py

```diff
@@ -517,19 +685,85 @@ def get(prefixes,
         urls = [(prefix_url, 0) for prefix_url in urllist]
 
     # exclude the non-existent files from loading
-    to_load = [(url, size) for url, size in urls if size is not None]
-    process_urls('download', to_load, verbose, num_workers)
+    to_load = [url for url, size in urls if size is not None]
+    sz_results = process_urls(dl_op, to_load, verbose, num_workers)
+    # We check if there is any access denied
+    is_denied = [sz == -ERROR_URL_ACCESS_DENIED for sz in sz_results]
+    if any(is_denied):
+        # Find the first one to return that as an error
+        for i, b in enumerate(is_denied):
+            if b:
+                exit(ERROR_URL_ACCESS_DENIED, to_load[i])
+    if not allow_missing:
+        is_missing = [sz == -ERROR_URL_NOT_FOUND for sz in sz_results]
+        if any(is_missing):
+            # Find the first one to return that as an error
+            for i, b in enumerate(is_missing):
+                if b:
+                    exit(ERROR_URL_NOT_FOUND, to_load[i])
     # Postprocess
     if verify:
-        verify_results(to_load, verbose=verbose)
+        # Verify only results with an actual size (so actual files)
+        verify_results([(url, sz) for url, sz in zip(to_load, sz_results)
+                        if sz != -ERROR_URL_NOT_FOUND], verbose=verbose)
 
+    idx_in_sz = 0
     if listing:
-        for url, size in urls:
-            if size is None:
+        for url, _ in urls:
+            sz = None
+            if idx_in_sz != len(to_load) and url.url == to_load[idx_in_sz].url:
+                sz = sz_results[idx_in_sz] if sz_results[idx_in_sz] >= 0 else None
+                idx_in_sz += 1
+            if sz is None:
+                # This means that either the initial url had a None size or
+                # that after loading, we found a None size
                 print(format_triplet(url.url))
             else:
                 print(format_triplet(url.prefix, url.url, url.local))
 
+@cli.command(help='Get info about files from S3')
+@click.option('--num-workers',
+              default=NUM_WORKERS_DEFAULT,
+              show_default=True,
+              help='Number of concurrent connections.')
+@click.option('--inputs',
+              type=click.Path(exists=True),
+              help='Read input prefixes from the given file.')
+@click.option('--verbose/--no-verbose',
+              default=True,
+              show_default=True,
+              help='Print status information on stderr.')
+@click.option('--listing/--no-listing',
+              default=False,
+              show_default=True,
+              help='Print S3 URL -> local file mapping on stdout.')
+@click.argument('prefixes', nargs=-1)
+def info(prefixes,
```

Comment:
Is there a reason for not supporting `--recursive` for `info`?

Reply:
I can add it. I didn't add any info_recursive or info_all at the TL but we can do it if you think it is useful.